src/components/user/unavailability/unavailability.edgedb.repository.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common';
+import { Range } from 'edgedb';
+import { ID, PublicOf } from '~/common';
+import { e, RepoFor } from '~/core/edgedb';
+import {
+  CreateUnavailability,
+  Unavailability,
+  UpdateUnavailability,
+} from './dto';
+import { UnavailabilityRepository } from './unavailability.repository';
+
+@Injectable()
+export class UnavailabilityEdgeDBRepository
+  extends RepoFor(Unavailability, {
+    hydrate: (unavailability) => ({
+      ...unavailability['*'],
+      start: e.assert_exists(e.range_get_lower(unavailability.dates)),
+      end: e.date_range_get_upper(unavailability.dates),
+    }),
+  }).customize((cls) => {
+    return class extends cls {
+      async create(input: CreateUnavailability) {
+        const user = e.cast(e.User, e.uuid(input.userId));
+        const unavailability = e.insert(e.User.Unavailability, {
+          description: input.description,
+          dates: new Range(input.start, input.end, true, true),
+        });
+        const query = e.update(user, () => ({
+          set: {
+            unavailabilities: { '+=': unavailability },
+          },
+        }));
+        await this.db.run(query);
+        const results = e.assert_exists(
+          e.select(e.User.Unavailability, (u) => ({
+            filter_single: { id: unavailability.id },
+            ...u,
+            start: e.assert_exists(e.range_get_lower(u.dates)),
+            end: e.assert_exists(e.range_get_upper(u.dates)),
+          })),
+        );
+        return await this.db.run(results);
```

Comment:
@CarsonF - Would this be possible now that the start/end are up in the hydrate?

```suggestion
          e.select(e.User.Unavailability, (u) => ({
            filter_single: { id: unavailability.id },
          })),
        );
        return await this.db.run(results, this.hydrate);
```

Reply:
Yes it would. But you also shouldn't be running two queries.
Try
```ts
const updatedUser = e.update(user, () => ({
  set: {
    unavailabilities: { '+=': unavailability },
  },
}));

// Might need to wrap in assert single/exists
const created = e.select(updatedUser.unavailabilities, (u) => ({
  filter: e.op(u.id, '=', unavailability.id)
}));

const query = e.select(created, this.hydrate);
```

Reply:
@CarsonF, I tried to get it working but for some reason it seems like the hydrate is not picking up/adding the additional properties.  I pushed the code so it's easier for you to review.

Reply:
ok, thank you!